src/mavlink-router/endpoint.cpp

```diff
@@ -70,20 +70,20 @@ int Endpoint::handle_read()
 {
     int target_sysid, target_compid, r;
     uint8_t src_sysid, src_compid;
+    uint32_t msg_id;
     struct buffer buf{};
 
-    while ((r = read_msg(&buf, &target_sysid, &target_compid, &src_sysid, &src_compid)) > 0)
+    if ((r = read_msg(&buf, &target_sysid, &target_compid, &src_sysid, &src_compid, &msg_id)) > 0)
```

Comment:
should continue to be `while`, or put in a separate commit with justification

Reply:
Reverted - will create a new cleanup PR with justification.